### ISSUE TITLE
Unhelpful Subword Assignment Error (in Firrtl)

### ISSUE THREAD

**Issue by jackkoenig** [freechipsproject/chisel3#399]
See http://stackoverflow.com/questions/40950073/chisel-3-assignment-to-bit-range

```scala
class TestX extends Module
{
    val io = IO(new Bundle {
        val a = Output(UInt(width=2))
    })

    io.a(1, 0) := UInt(0)
}
```

Gives error:
`Error: [module TestX] Expression T_4 is used as a FEMALE but can only be used as a MALE.`

We should probably try to catch this in Chisel

**Comment by jackkoenig** [comment]
Also we should probably document this limitation

**Comment by ducky64** [comment]
Subword assignments not supported.

It may be possible to emit a better error by adding a SourceOnlyBinding or something, so assigning to it in Chisel frontend will give the user a helpful error.

**Comment by ducky64** [comment]
This will be implemented after / with #440
